AirCasting/Map/GoogleMapView.swift

```diff
@@ -29,6 +37,9 @@ struct GoogleMapView: UIViewRepresentable {
         let mapView = GMSMapView.map(withFrame: frame,
                                      camera: startingPoint)
         
+        mapView.delegate = delegateRelay
```

Comment:
but aren't we (`GoogleMapView`) holding it strongly tho by using a property for it? So it's guaranteed to be alive as long as `GoogleMapView` is alive? 🤔 And if the whole view it's contained in reloads than we're just creating all of this from scratch and still are ok? IDK, will give it some more testing perhaps after adjusting for switching streams

Reply:
Well, my thinking was that delegate's in the struct which is created new every time we refresh the view. I did some testing and found two things. First is that the delegateRelay is indeed created new every time we refresh any of the properties (threshold, locationEnabled or pathpoints) but the new delegate isn't used anywhere.
And the second thing is that I found a memory leak. It might be helpful for you to know that ;D
![Screenshot 2021-08-06 at 09 09 30](https://user-images.githubusercontent.com/46443841/128489831-253707eb-8ea1-47cb-ae87-80fe67aacc30.png)

Reply:
Yeaaaah, valid point. Now I see. Will need to use coordinator here obviously. Good catch! SwiftUI and its gotchas... (I mean - it's a bit counterintuitive that this struct is being recreated just like any other view, *but* the sequence of calls to makeUIView/updateUIView is very object-like)

Reply:
I have an idea for a temporary solution if you want. But not sure if it's valid. We could add `mapView.delegate = delegateRelay` to the function updating the view :)

Reply:
Will check out the proper solution today (after pulling changes from develop now the graph/map is popping right back so some more issues to fix...). If I encounter problems with it I will probably do so :) thanks for the suggestion!